impl/src/main/java/org/jboss/weld/event/TransactionalObserverNotifier.java

```diff
@@ -73,8 +77,33 @@ protected TransactionalObserverNotifier(String contextId, TypeSafeObserverResolv
             // Transaction is not active - no deferred notifications
             notifySyncObservers(observers, event, metadata);
         } else {
-            for (ObserverMethod<? super T> observer : observers) {
-                deferNotification(event, metadata, observer);
+            try {
+                for (ObserverMethod<? super T> observer : observers) {
+                    deferNotification(event, metadata, observer);
```

Comment:
Theoretically, it's possible that some registrations pass. So in the catch block we should only process the observers for which the registration failed. E.g. create a copy of observers outside the catch block, remove each observer for which the registration passed and then process the copy in the catch block.